pkg/ext/extension_apiserver.go

```diff
@@ -148,22 +148,30 @@ func NewExtensionAPIServer(ctx context.Context, wranglerContext *wrangler.Contex
 	case "true":
 		logrus.Info("creating imperative extension apiserver resources")
 
-		sniProvider, err := certForCommonName(fmt.Sprintf("%s.%s.svc", TargetServiceName, Namespace))
+		sniProvider, err := NewSNIProviderForCname(
```

Comment:
Does this work work correct in a HA rancher environment or do all replicas fight to update the certs once it expires?

A quick glance at dynamiclistener code and it looks like they would also all compete on updating the certs, so I think it's acceptable to do it here as well, and fix it only if it becomes a problem. Thoughts?

Reply:
Yeah they do fight to fix, but once one updates the rest will fail. It bothered me too and I did the same initial look through dynamic listener and decided it would be super good enough.

Don't know how much we'll benefit from doing some kind of leader election here.